src/sidebar/components/moderation/ModerationControl.tsx

```diff
@@ -0,0 +1,51 @@
+import { ModerationStatusSelect } from '@hypothesis/annotation-ui';
+
+import type { Annotation } from '../../../types/api';
+import { useSidebarStore } from '../../store';
+import ModerationStatusBadge from './ModerationStatusBadge';
+
+export type ModerationControlProps = {
+  annotation: Annotation;
+  /** Classes to apply to the ModerationStatusBadge in case it's rendered */
+  badgeClasses?: string | string[];
+};
+
+/**
+ * Render the proper moderation control/s for provided annotation.
+ * - ModerationStatusSelect for moderators of pre-moderated groups.
+ * - ModerationStatusBadge for own annotations in any group
+ */
+export default function ModerationControl({
+  annotation,
+  badgeClasses,
+}: ModerationControlProps) {
+  const store = useSidebarStore();
+  const currentUserId = store.profile().userid;
+  const groupIsPreModerated = store.focusedGroup()?.pre_moderated;
+  const currentUserIsModerator = annotation.actions?.includes('moderate');
+
+  // We want to show moderation controls only if current user is a moderator or
+  // the annotation's author
+  const moderationStatus =
+    currentUserIsModerator || annotation.user === currentUserId
+      ? annotation?.moderation_status
+      : undefined;
+
+  if (!moderationStatus) {
+    return null;
+  }
+
+  // We don't want to show moderation selects to moderators of non-pre-moderated
+  // groups, to avoid cluttering the UI with all the extra controls
```

Comment:
I think it is probably worth expanding this to say that we don't want to clutter the UI with extra controls when the majority of annotations will have the "Approved" status. We may offer this control under a menu instead in future.